docs/src/main/asciidoc/mongodb-panache-guide.adoc

```diff
@@ -0,0 +1,525 @@
+
+Imagine the following entity:
+
+[source,java]
+--
+public class Person extends PanacheMongoEntity {
+    @BsonProperty("lastname")
+    public String name;
+    public LocalDate birth;
+    public Status status;
+
+    public static Person findByNameWithPanacheQLQuery(String name){
+      return find("name", name).firstResult();
+    }
+
+    public static Person findByNameWithNameQuery(String name){
+        return find("{'lastname': ?1}", name).firstResult();
+    }
+}
+--
+
+Both `findByNameWithPanacheQLQuery()` and `findByNameWithNameQuery()` methods will return the same result but query written in PanacheQL
+will use the entity field name: `name`, and native query will use the MongoDB field name: `lastname`.
+
+== Query projection
+
+Query projection can be done with the `project()` method on the `PanacheQuery` object that is returned by the `find()` method.
+You can use it to restrict which fields will be returned by the database.
+Note that the ID field of your entity (the one corresponding to the MongoDB `_id` field) will always be returned.
+
+[source,java]
+--
+// only 'name' and 'id' fields will be returned
+PanacheQuery<Person> query = Person.find("status ", Status.Alive).project("name");
+PanacheQuery<Person> query = Person.find("{'status': ?1}", Status.Alive).project("name");
+--
+
+== The DAO/Repository option
+
+Look, we get it: you have a love/hate relationship with DAOs/Repositories but you can't live without them. We don't judge, we know life is tough and we've got you covered.
+
+If you want to have Repositories, you can get the exact same convenient methods injected in your Repository by making it
+implement `PanacheMongoRepository`:
+
+[source,java]
+--
+@ApplicationScoped
+public class PersonRepository implements PanacheMongoRepository<Person> {
+
+   // put your custom logic here as instance methods
+
+   public Person findByName(String name){
+     return find("name", name).firstResult();
+   }
+
+   public List<Person> findAlive(){
+     return list("status", Status.Alive);
+   }
+
+   public void deleteLoics(){
+     delete("name", "Loïc");
+  }
+}
+--
+
+Absolutely all the operations that are defined on `PanacheMongoEntityBase` are available on your DAO, so using it
+is exactly the same except you need to inject it:
+
+[source,java]
+--
+@Inject
+PersonRepository personRepository;
+
+@GET
+public long count(){
+  return personRepository.count();
+}
+--
+
+So if Repositories are your thing, you can keep doing them. Even with repositories, you can keep your entities as
+subclasses of `PanacheMongoEntity` in order to get the ID and public fields working, but you can even skip that and
+go back to specifying your ID and using getters and setters if that's your thing. We're not judging.
+
+== Transactions
+
+MongoDB offers ACID transactions since version 4.0. MongoDB with Panache doesn't provide support for them.
```

Comment:
Maybe make it a warning?

Reply:
done
I hope it can be done soon as Narayana should be able to provide support for it